rose/server/player_test.py

```diff
@@ -1,15 +1,75 @@
+from rose.common import actions, config
 from . import player
 
 
-def test_in_lane():
-    p = player.Player("A", car=0, lane=0)
-    for x in (0, 1, 2):
-        p.x = x
-        assert p.in_lane()
+def test_player_initialization():
+    player1 = player.Player("John", 1, 1)
 
+    assert player1.name == "John"
+    assert player1.car == 1
+    assert player1.lane == 1
+    # player.Player x value should be middle of it's lane
+    assert player1.x == player1.lane * config.cells_per_player + 1
+    assert player1.y == config.matrix_height // 3 * 2
+    assert player1.action == actions.NONE
+    assert player1.response_time is None
+    assert player1.score == 0
 
-def test_not_in_lane():
-    p = player.Player("A", car=0, lane=1)
-    for x in (0, 1, 2):
-        p.x = x
-        assert not p.in_lane()
+
+def test_player_reset():
+    player1 = player.Player("John", 1, 1)
+
+    player1.score = 50  # Modify player to make sure reset works
+    player1.reset()
+    assert player1.x == player1.lane * config.cells_per_player + 1
+    assert player1.y == config.matrix_height // 3 * 2
+    assert player1.action == actions.NONE
+    assert player1.response_time is None
+    assert player1.score == 0
+
+
+def test_player_in_lane_true():
+    player1 = player.Player("John", 1, 1)
+
+    # This test assumes the player is in their lane after reset.
+    # So, we test the default state after initialization.
+    assert player1.in_lane() is True
+
+
+def test_player_not_in_lane():
+    player1 = player.Player("John", 1, 1)
+
+    # Modify player's position to be out of their lane
+    other_lane = (player1.lane + 1) % config.max_players
+    player1.x = other_lane * config.cells_per_player + 1
```

Comment:
Would be nice to test that all 3 possible cells in the other lane are handle correctly, something like:

    base = other_lane * other_lane * config.cells_per_player
    for offset in range(config.cells_per_player):
        player.x = base + offset
        assert not player1.in_lane()

Reply:
done :-)